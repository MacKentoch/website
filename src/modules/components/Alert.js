import styled from 'styled-components'
import { darken, lighten } from 'polished'
import theme from 'style/theme'

const Alert = styled.div`
  padding: 10px;
  border-radius: 4px;
  background-color: ${props => lighten(0.3, theme.colors[props.ui])};
  color: ${props => darken(0.2, theme.colors[props.ui])};
`

Alert.defaultProps = {
  ui: 'danger',
}

export default Alert
