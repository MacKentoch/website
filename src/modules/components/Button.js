import styled from 'styled-components'
import { darken } from 'polished'
import theme from 'style/theme'

const Button = styled.button`
  border-radius: 3px;
  display: inline-block;
  height: 38px;
  padding: 0 30px;
  text-align: center;
  font-size: 16px;
  line-height: 38px;
  white-space: nowrap;;
  color: white;
  cursor: pointer;
  transition: background-color 200ms;
  text-decoration: none;
  border: 0;
  background-color: ${theme.colors.primary};
  &:hover, &:focus {
    background-color: ${darken(0.2, theme.colors.primary)};
    outline: none;
  }
`

export default Button
