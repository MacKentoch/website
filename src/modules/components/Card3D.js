import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import offset from 'dom-helpers/query/offset'

const CardContainer = styled.div`
  transform-style: preserve-3d;
  perspective: 1000px;
  width: ${props => props.width}px;
  height: ${props => props.height}px;

  .anchor {
    display: block;
    text-decoration: none;
    position: relative;
    border-radius: 3px;
    transition: transform .4s cubic-bezier(0, 0, .2, 1),
      filter .15s cubic-bezier(0, 0, .2, 1);
    width: 100%;
    height: 100%;
    transform: scale(1) rotateY(1e-07deg) rotateX(0) rotateZ(0deg);
  }

  .background {
    content: "";
    display: block;
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    top: 0;
    border-radius: 3px;
    overflow: hidden;
    pointer-events: none;
    box-shadow: 0 10px 25px 0 rgba(50, 94, 128, .2);
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
  }

  .glare {
    content: "";
    display: block;
    position: absolute;
    mix-blend-mode: overlay;
    background-image: radial-gradient(
      ellipse closest-side,
      hsla(0, 0%, 100%, .45),
      hsla(0, 0%, 100%, 0)
    );
    bottom: 0;
    left: -50%;
    right: -50%;
    top: -50%;
    height: 125%;
    will-change: transform;
    transition: transform .4s cubic-bezier(0, 0, .2, 1);
    opacity: .999999;
    z-index: 3;
    transform: translate(1px, 0px);
  }

  .inner {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    pointer-events: none;
  }
`

class Card3D extends Component {
  static propTypes = {
    background: PropTypes.string.isRequired,
    children: PropTypes.node,
    height: PropTypes.number.isRequired,
    width: PropTypes.number.isRequired,
  }

  state = {
    style: {},
    glareStyle: {},
  }

  handleMouseMove = event => {
    const elementOffset = offset(event.currentTarget)
    const w = elementOffset.width
    const h = elementOffset.height
    const sensitivity = 2000 / Math.max(w, h)
    const currentX = Math.round(event.pageX - elementOffset.left)
    const currentY = Math.round(event.pageY - elementOffset.top)
    const cx = (w / 2 - currentX) / w * 2
    const cy = (h / 2 - currentY) / h * 2
    const rx = cy * sensitivity
    const ry = -cx * sensitivity
    const tx = cx * w / 5
    const ty = cy * h / 5

    this.setState({
      style: {
        transform: `scale(1.05) rotateY(${ry}deg) rotateX(${rx}deg) rotateZ(0deg)`,
      },
      glareStyle: {
        transform: `translate(${tx}px, ${ty}px)`,
      },
    })
  }

  handleMouseLeave = () => {
    this.setState({
      style: {},
      glareStyle: {},
    })
  }

  render() {
    return (
      <CardContainer width={this.props.width} height={this.props.height}>
        <div
          className="anchor"
          style={this.state.style}
          onMouseMove={this.handleMouseMove}
          onMouseLeave={this.handleMouseLeave}
        >
          <div
            className="background"
            style={{ backgroundImage: this.props.background }}
          >
            <div className="glare" style={this.state.glareStyle} />
          </div>
          <div className="inner">
            {this.props.children}
          </div>
        </div>
      </CardContainer>
    )
  }
}

export default Card3D
