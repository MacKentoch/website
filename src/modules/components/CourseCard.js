import React from 'react'
import styled from 'styled-components'
import { transparentize, darken } from 'polished'
import Markdown from 'modules/components/Markdown'

const Container = styled.div`
  background-image: linear-gradient(
    -180deg,
    ${props => transparentize(0.95, props.color)} 0%,
    ${props => transparentize(0.9, props.color)} 100%
  );
  box-shadow: 0 7px 14px 0 rgba(93, 60, 50, 0.10),
    0 3px 6px 0 rgba(0, 0, 0, 0.08);
  border-radius: 4px;
  padding: 20px;
`

const Title = styled.h4`
  font-weight: 600;
  font-size: 20px;
  line-height: 24px;
  margin: 0;
  color: ${props => darken(0.3, props.color)};
`

const CourseCard = ({ title, outline, path }) =>
  <Container color={path.color}>
    <Title color={path.color}>
      {title}
    </Title>
    <Markdown source={outline} />
  </Container>

export default CourseCard
