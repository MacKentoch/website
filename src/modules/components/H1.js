import styled from 'styled-components'

const H1 = styled.h1`
  font-size: 40px;
  line-height: 1.2;
  margin: 10px 0;
  font-weight: 400;
  @media (min-width: 700px) {
    font-size: 50px;
  }
`

export default H1
