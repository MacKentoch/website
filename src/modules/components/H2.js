import styled from 'styled-components'

const H2 = styled.h2`
  font-size: 26px;
  text-align: center;
  line-height: 1.4;
  font-weight: 300;
  margin: 20px 0;
  @media (min-width: 700px) {
    font-size: 40px;
    text-align: left;
  }
`

export default H2
