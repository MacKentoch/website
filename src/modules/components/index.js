export { default as Alert } from './Alert'
export { default as Button } from './Button'
export { default as H1 } from './H1'
export { default as H2 } from './H2'
export { default as Input } from './Input'
export { default as Textarea } from './Textarea'
export { default as OSXWindow } from './OSXWindow'
export { default as Wrapper } from './Wrapper'
