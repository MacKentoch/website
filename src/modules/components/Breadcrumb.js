import React from 'react'
import styled from 'styled-components'
import { RouterLink } from 'modules/components/Link'
import JsonLd from 'modules/components/JsonLd'
import { breadcrumbLd } from 'client/linkedData'
import { completeUrl } from 'modules/urlUtil'
import theme from 'style/theme'

const Container = styled.div`
  font-size: 20px;
  line-height: 24px;
  font-weight: 300;
  padding-bottom: 15px;
  margin-bottom: 15px;
  border-bottom: 1px solid ${theme.colors.grayLight};
`

const Breadcrumb = ({ links }) =>
  <Container>
    {links.reduce((elements, link, index) => {
      if (index !== links.length - 1) {
        elements.push(
          <RouterLink key={link.url} to={link.url}>
            {link.name}
          </RouterLink>,
        )
        elements.push(' > ')
      } else {
        elements.push(link.name)
      }

      return elements
    }, [])}
    <JsonLd>
      {breadcrumbLd({
        links: links.map(link => ({ ...link, url: completeUrl(link.url) })),
      })}
    </JsonLd>
  </Container>

export default Breadcrumb
