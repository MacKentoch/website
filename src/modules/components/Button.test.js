import React from 'react'
import { shallowWithTheme } from 'test/testUtils'
import components from './'

describe('Button', () => {
  it('should render a button', () => {
    expect(
      shallowWithTheme(<components.Button>Hello</components.Button>).shallow(),
    ).toMatchSnapshotWithGlamor()
  })

  it('should render a specific component', () => {
    expect(
      shallowWithTheme(<components.Button component="a">Hello</components.Button>).shallow(),
    ).toMatchSnapshotWithGlamor()
  })
})
