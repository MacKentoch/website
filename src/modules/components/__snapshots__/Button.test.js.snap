// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a button 1`] = `
.css-tf2nfx,
[data-css-tf2nfx] {
  -webkit-transition: background-color 200ms;
  border-radius: 3px;
  height: 38px;
  padding: 0 30px;
  text-align: center;
  font-size: 16px;
  line-height: 38px;
  white-space: nowrap;
  display: inline-block;
  cursor: pointer;
  transition: background-color 200ms;
  text-decoration: none;
  border: 0;
  background-color: #BD4932;
  children: Hello;
  color: white;
}

.css-tf2nfx:hover,
[data-css-tf2nfx]:hover,
.css-tf2nfx[data-simulate-hover],
[data-css-tf2nfx][data-simulate-hover] {
  background-color: #6c2a1d;
  outline: none;
}

.css-tf2nfx:focus,
[data-css-tf2nfx]:focus,
.css-tf2nfx[data-simulate-focus],
[data-css-tf2nfx][data-simulate-focus] {
  background-color: #6c2a1d;
  outline: none;
}

<button
  className="css-tf2nfx"
>
  Hello
</button>
`;

exports[`Button should render a specific component 1`] = `
.css-10m27ti,
[data-css-10m27ti] {
  -webkit-transition: background-color 200ms;
  border-radius: 3px;
  height: 38px;
  padding: 0 30px;
  text-align: center;
  font-size: 16px;
  line-height: 38px;
  white-space: nowrap;
  display: inline-block;
  cursor: pointer;
  transition: background-color 200ms;
  text-decoration: none;
  border: 0;
  background-color: #BD4932;
  component: a;
  children: Hello;
  color: white;
}

.css-10m27ti:hover,
[data-css-10m27ti]:hover,
.css-10m27ti[data-simulate-hover],
[data-css-10m27ti][data-simulate-hover] {
  background-color: #6c2a1d;
  outline: none;
}

.css-10m27ti:focus,
[data-css-10m27ti]:focus,
.css-10m27ti[data-simulate-focus],
[data-css-10m27ti][data-simulate-focus] {
  background-color: #6c2a1d;
  outline: none;
}

<a
  className="css-10m27ti"
>
  Hello
</a>
`;
